glean-core/src/common_metric_data.rs

```diff
@@ -64,6 +66,9 @@ pub struct CommonMetricData {
     ///
     /// Disabled metrics are never recorded.
     pub disabled: bool,
+    /// Dynamic label. These need to be looked up in the database to make sure
+    /// we aren't storing too many. (Static labels are stored as part of name).
+    pub dynamic_label: Option<String>,
```

Comment:
Why can't this live in the `LabeledMetricType`?

Reply:
It needs to exist in every metric type that can be labeled.  So on `counter`, not `labeled_counter`.

`LabeledMetricType` is basically a factory for a regular metric type, if that helps.

The `labeled_counter` creates a specific copy of a `counter` with the label value stored so that later when the actual recording happens in a coroutine we can do the validation and correction of that label.

Reply:
I see. Much clearer now, thanks for bearing with me on this. Can you integrate the comment above this line with what you said here?